electricitymap/contrib/capacity_parsers/EIA.py

```diff
@@ -0,0 +1,105 @@
+import calendar
+from datetime import datetime
+from logging import getLogger
+from typing import Dict, Union
+
+import pandas as pd
+from requests import Response, Session
+
+from electricitymap.contrib.config import ZoneKey
+from parsers.EIA import REGIONS
+from parsers.lib.utils import get_token
+
+logger = getLogger(__name__)
+
+CAPACITY_URL = "https://api.eia.gov/v2/electricity/operating-generator-capacity/data/?frequency=monthly&data[0]=nameplate-capacity-mw&facets[balancing_authority_code][]={}"
+API_KEY = get_token("EIA_KEY")
+
+US_ZONES = {key: value for key, value in REGIONS.items() if key.startswith("US-")}
+TECHNOLOGY_TO_MODE = {
+    "All Other": "unknown",
+    "Batteries": "battery storage",
+    "Coal Integrated Gasification Combined Cycle": "coal",
+    "Conventional Hydroelectric": "hydro",
+    "Conventional Steam Coal": "coal",
+    "Flywheels": "battery storage",
+    "Geothermal": "geothermal",
+    "Hydroelectric Pumped Storage": "hydro storage",
+    "Landfill Gas": "biomass",
+    "Municipal Solid Waste": "biomass",  # or unknown?
+    "Natural Gas Fired Combined Cycle": "gas",
+    "Natural Gas Fired Combustion Turbine": "gas",
+    "Natural Gas Internal Combustion Engine": "gas",
+    "Natural Gas Steam Turbine": "gas",
+    "Natural Gas with Compressed Air Storage": "gas",
+    "Nuclear": "nuclear",
+    "Offshore Wind Turbine": "wind",
+    "Onshore Wind Turbine": "wind",
+    "Other Gases": "unknown",
+    "Other Natural Gas": "gas",
+    "Other Waste Biomass": "biomass",
+    "Petroleum Coke": "coal",
+    "Petroleum Liquids": "oil",
+    "Solar Photovoltaic": "solar",
+    "Solar Thermal with Energy Storage": "solar",
+    "Solar Thermal without Energy Storage": "solar",
+    "Wood/Wood Waste Biomass": "biomass",
+}
+
+
+def format_capacity(df: pd.DataFrame, target_datetime: datetime) -> dict:
+    df = df.copy()
+    df = df.loc[df["statusDescription"] == "Operating"]
+    df["mode"] = df["technology"].map(TECHNOLOGY_TO_MODE)
+    df_aggregated = df.groupby(["mode"])[["nameplate-capacity-mw"]].sum().reset_index()
+    capacity_dict = {}
+    for mode in df_aggregated["mode"].unique():
+        mode_dict = {}
+        mode_dict["value"] = float(
+            df_aggregated.loc[df_aggregated["mode"] == mode][
+                "nameplate-capacity-mw"
+            ].sum()
+        )
+        mode_dict["source"] = "EIA"
```

Comment:
```suggestion
        mode_dict["source"] = SOURCE
```
I would recommend following the same structure as with other parsers.